support-frontend/assets/pages/digital-subscription-checkout/components/datePicker/datePicker.jsx

```diff
@@ -0,0 +1,256 @@
+// @flow
+
+import React, { Component } from 'react';
+import DayPicker, { DateUtils } from 'react-day-picker';
+import 'react-day-picker/lib/style.css';
+import { css } from '@emotion/core';
+import { space } from '@guardian/src-foundations';
+import { formatMachineDate } from 'helpers/dateConversions';
+import calendarIcon from './calendarIcon.png';
+import { type Option } from 'helpers/types/option';
+import { monthText } from 'pages/paper-subscription-checkout/helpers/subsCardDays';
+import { Input } from 'components/forms/input';
+import { withLabel } from 'hocs/withLabel';
+import { ThemeProvider } from 'emotion-theming';
+import { Button, buttonDefault } from '@guardian/src-button';
+import { Error } from 'components/forms/customFields/error';
+
+import './styles.scss';
+
+const calendarIconContainer = css`
+  padding: 0;
+  border: none;
+  width: 35px;
+  height: 35px;
+  align-self: flex-end;
+  margin: 0 0 ${space[1]}px ${space[4]}px ;
+`;
+
+const iconImage = css`
+  width: 100%;
+  height: 100%;
+`;
+
+const startDateGroup = css`
+  display: flex;
+  flex-direction: column;
+`;
+
+const startDateFields = css`
+  display: inline-flex;
+`;
+
+const inputLayout = css`
+  width: 25%;
+`;
+
+const inputLayoutWithMargin = css`
+  width: 25%;
+  margin-right: ${space[3]}px;
+`;
+
+const validationButton = css`
+  margin-top: ${space[5]}px;
+`;
+
+const marginTop = css`
+  margin-top: ${space[5]}px;
+`;
+
+type PropTypes = {
+  // eslint-disable-next-line react/no-unused-prop-types
+  value: string | null,
+  onChange: Function,
+  // eslint-disable-next-line react/no-unused-prop-types
+  id: Option<string>,
+}
+
+type StateTypes = {
+  day: string,
+  month: string,
+  year: string,
+  showCalendar: boolean,
+  dateError: string | null,
+  dateValidated: boolean | null,
+}
+
+const InputWithLabel = withLabel(Input);
+
+class DatePickerFields extends Component<PropTypes, StateTypes> {
```

Comment:
This would be a really good candidate for adding to Storybook. It would also have been a good to create a PR specifically for this component I think.

Reply:
Also some unit tests might be helpful

Reply:
Wow, yes - the storybook point is a missed opportunity. I wish I had thought of that. And you're absolutely right about tests. 👍